server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -38,8 +38,8 @@
  */
 public class RemoteClusterPortSettings {
 
-    public static final TransportVersion TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_8_0;
-    public static final Version VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = Version.V_8_8_0;
+    public static final TransportVersion TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_9_0;
+    public static final Version VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = Version.V_8_9_0;
```

Comment:
Please let me know if you want me to remove this constant in this PR. 
Relates: https://github.com/elastic/elasticsearch/pull/95442#discussion_r1173658588

Reply:
I removed this constant to in favor of the transport one in this PR because I don't want to introduce yet another 8.9 Version constant to gate specialized API keys in ApiKeyService.